qa/common/trace_summary.py

```diff
@@ -50,54 +53,55 @@ def summarize(traces):
         for ts in trace["timestamps"]:
             timestamps[ts["name"]] = ts["ns"]
 
-        key = (trace["model_name"], trace["model_version"])
-        if key not in model_count_map:
-            model_count_map[key] = 0
-            model_span_map[key] = dict()
+        if ("request handler start" in timestamps) and ("request handler end" in timestamps):
+            key = (trace["model_name"], trace["model_version"])
+            if key not in model_count_map:
+                model_count_map[key] = 0
+                model_span_map[key] = dict()
+
+            model_count_map[key] += 1
+            if ("http recv start" in timestamps) and ("http send end" in timestamps):
+                add_span(model_span_map[key], timestamps,
+                         "http infer", "http recv start", "http send end")
+                add_span(model_span_map[key], timestamps,
+                         "http recv", "http recv start", "http recv end")
+                add_span(model_span_map[key], timestamps,
+                         "http send", "http send start", "http send end")
+            elif ("grpc wait/read start" in timestamps) and ("grpc send end" in timestamps):
+                add_span(model_span_map[key], timestamps,
+                         "grpc infer", "grpc wait/read start", "grpc send end")
```

Comment:
`"grpc wait/read start"` is recorded when a GRPC request is created, so GRPC infer time here will be longer than it actually is. Do we need to point this out somewhere?

Reply:
Unfortunately, GRPC doesn't provide a hook where we can get just the request read timestamp. I will document the tracer in general and include this type of information.